tests/streamlit_ui_tests/set_column_formula_tests.spec.ts

```diff
@@ -0,0 +1,288 @@
+import { FrameLocator, Page, expect, test } from '@playwright/test';
+import { 
+    getMitoFrameWithTypeCSV, 
+    createNewColumn, 
+    setFormulaUsingCellEditor, 
+    getCellAtRowIndexAndColumnName, 
+    getValuesInColumn, 
+    getMitoFrameWithTestCSV, 
+    awaitResponse, 
+    toggleEditEntireColumn 
+} from './utils';
+
+
+test('Set constant formula to new column using cell editor', async ({ page }) => {
+    const columnHeader = 'Column4';
+
+    const mito = await getMitoFrameWithTypeCSV(page);
+    await createNewColumn(page, mito, 3, columnHeader);
+    await setFormulaUsingCellEditor(page, mito, columnHeader, '=5');
+
+    // Check that the value in columnHeader are '5'
+    const cellValues = await getValuesInColumn(mito, columnHeader);
+    expect(cellValues).toEqual(Array(cellValues.length).fill('5'));
+
+    // Test that reopening the cell editor displays the formula
+    const cell = await getCellAtRowIndexAndColumnName(mito, 0, columnHeader);
+    await cell.dblclick();
+    await expect(mito.locator('#cell-editor-input')).toHaveValue('=5');
+
+    // Check that the formula bar also displays the formula
+    expect(await (mito.locator('.formula-bar-formula')).textContent()).toEqual('=5');
+
+    // Test that using escape closes the cell editor
+    await expect(mito.locator('#cell-editor-input')).toBeVisible();
+    await mito.locator('#cell-editor-input').press('Escape');
+    await expect(mito.locator('#cell-editor-input')).not.toBeVisible();
+});
+
+test('Set constant formula to existing column using cell editor', async ({ page }) => {
+    const columnHeader = 'Column1';
+
+    const mito = await getMitoFrameWithTypeCSV(page);
+    await setFormulaUsingCellEditor(page, mito, columnHeader, '=5');
+
+    // Check that the value in columnHeader are '5'
+    const cellValues = await getValuesInColumn(mito, columnHeader);
+    expect(cellValues).toEqual(Array(cellValues.length).fill('5'));
```

Comment:
Wait I'm confused about this one. What exactly is this line doing?

Reply:
This is just checking that the values in column `columnHeader` are all 5. Which they should be because I set the formula to =5.

Reply:
Ohhh wow ok that's a syntax that I was not aware of